src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1274,6 +1276,75 @@ public IEnumerable<Tombstone> GetTombstonesFrom(DocumentsOperationContext contex
             }
         }
 
+        public IEnumerable<TombstoneIndexItem> GetTombstoneIndexItemsFrom(DocumentsOperationContext context, long etag, long start, long take)
+        {
+            var table = new Table(TombstonesSchema, context.Transaction.InnerTransaction);
+
+            // ReSharper disable once LoopCanBeConvertedToQuery
+            foreach (var result in table.SeekForwardFrom(TombstonesSchema.FixedSizeIndexes[AllTombstonesEtagsSlice], etag, start))
+            {
+                if (take-- <= 0)
+                    yield break;
+
+               
+                yield return TableValueToTombstoneIndexItem(context, ref result.Reader);
+            }
+        }
+
+        public IEnumerable<TombstoneIndexItem> GetTombstoneIndexItemsFrom(DocumentsOperationContext context, string collection, long etag, long start, long take)
+        {
+            string tableName;
+
+            if (collection == AttachmentsStorage.AttachmentsTombstones ||
+                collection == RevisionsStorage.RevisionsTombstones)
+            {
+                tableName = collection;
+            }
+            else
+            {
+                var collectionName = GetCollection(collection, throwIfDoesNotExist: false);
+                if (collectionName == null)
+                    yield break;
+
+                tableName = collectionName.GetTableName(CollectionTableType.Tombstones);
+            }
+
+            var table = context.Transaction.InnerTransaction.OpenTable(TombstonesSchema, tableName);
+
+            if (table == null)
+                yield break;
+
+            // ReSharper disable once LoopCanBeConvertedToQuery
+            foreach (var result in table.SeekForwardFrom(TombstonesSchema.FixedSizeIndexes[CollectionEtagsSlice], etag, start))
+            {
+                if (take-- <= 0)
+                    yield break;
+
+                yield return TableValueToTombstoneIndexItem(context, ref result.Reader);
+            }
+        }
+
+        private TombstoneIndexItem TableValueToTombstoneIndexItem(JsonOperationContext context, ref TableValueReader tvr)
+        {
+            if (tvr.Pointer == null)
+                return null;
+
+            var tableType = *(Tombstone.TombstoneType*)tvr.Read((int)TombstoneTable.Type, out int _);
+            var type = tableType switch
+            {
+                Tombstone.TombstoneType.Document => IndexItemType.Document,
+                Tombstone.TombstoneType.Counter => IndexItemType.Counters,
+                _ => IndexItemType.None
```

Comment:
We default to `None` because, at this point, we only handling documents. Other tombstone types exist, but they are currently not relevant for the specific operation (we skip other types, so we shouldn't throw)

Reply:
ok